enterprise/frontend/src/embedding/data-picker/DataSelector/DataSelector.jsx

```diff
@@ -409,8 +363,6 @@ export class UnconnectedDataSelector extends Component {
       this.state.selectedDataBucketId === DATA_BUCKET.MODELS
     ) {
       await this.switchToStep(DATABASE_STEP);
-    } else if (this.state.selectedTableId && steps.includes(FIELD_STEP)) {
```

Comment:
always false, since embedding data picker doesn't have `FIELD` step
 https://github.com/metabase/metabase/blob/b7e59e930ade75e768c61e52af336575d6d69489/enterprise/frontend/src/embedding/data-picker/DataSelector/DataSelector.jsx#L54